presto-parser/src/main/java/com/facebook/presto/sql/parser/ParsingOptions.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.parser;
+
+import static java.util.Objects.requireNonNull;
+
+public class ParsingOptions
+{
+    public enum FixedPointLiteralType
+    {
+        DOUBLE,
+        DECIMAL,
+        UNDEFINED
+    }
+
+    private final FixedPointLiteralType fixedPointLiteralType;
```

Comment:
"Fixed point" is not the right name for this. Fixed point vs floating point are ways of encoding fractional numbers. I'll try to think of something better.

Reply:
In the grammar it's called "decimal literal". What about

```
enum DecimalLiteralTreatment { AS_DOUBLE, AS_DECIMAL, REJECT }
```
?

Reply:
Sounds good